src/arx7.js

```diff
@@ -1,133 +1,199 @@
+
+    // Join channels
+    for(let i = 0; i < this.channels.length; i++) {
+      log(`Joining ${this.channels[i]}`);
 
-// Praise the Creator
-client.addListener('join', (channel, nick, message) => {
-  if (nick === 'Desch' && channel == '#arx-7') {
-    client.say(channel, 'Hello Master.');
+      if (this.config.channels[i].key) {
+        this.client.join(`${this.channels[i]} ${this.config.channels[i].key}`);
+      }
+      else {
+        this.client.join(this.channels[i]);
+      }
+    }
   }
-});
-
-// Custom Auto-Rejoin
-client.addListener('kick', (channel, nick, by, reason, message) => {
-  if (nick === client.nick) {
-    log(`Kicked from ${channel} by ${by}: ${reason}`);
-    setTimeout(() => {
-      log(`Rejoining ${channel}`);
-      client.join(channel);
-    }, 1000 * 3);
+
+  version(from, to, message) {
+    log(`CTCP request from ${from}`);
+    this.client.ctcp(from, 'notice', `VERSION ARX-7 v${pkg.version} (Bot)`);
   }
-});
 
-client.addListener('error', message => {
-  // Attempt to rejoin banned channels
-  if (message.command == 'err_bannedfromchan') {
-    log(`Banned from ${message.args[1]}. Rejoining in in 3 minutes`);
+  message(from, to, text, message) {
+    if (this.channels.includes(to)) {
+      this.commands.forEach(c => {
+        let plugin = c.constructor.name.toLowerCase();
+        let index = this.channels.indexOf(to);
+
+        if (this.config.channels[index].plugins.includes(plugin)) {
+          c.message(from, to, text, message);
+        }
+      });
+    }
 
-    // `()=>` ensures there is a delay; otherwise it continuously fires
-    setTimeout(() => client.join(message.args[1]), 1000 * 60 * 3);
+    // Handle Queries
+    if (to === this.client.nick) {
+      if (this.config.admins.includes(from)) {
+        this.handleAdminQuery(from, to, text, message);
+      }
+      else {
+        log(`Query from ${from}: ${text}`);
+        let admins = this.config.admins.join(', ');
+        this.client.say(from, `I'm a bot! Contact [${admins}] for help`);
+      }
+    }
   }
-  // Attempt to rejoin +k channels correctly
-  else if (message.command == 'err_badchannelkey') {
-    // Prevent infinite rejoin attempts
-    if (droppedChannels.includes(message.args[1])) {
-      log(`Incorrect password for ${message.args[1]}.`);
+
+  handleAdminQuery(from, to, text, message) {
+    log(`Admin Query from ${from}: ${text}`);
+    let usage = `[add|remove] [plugin] [channel] [password]`;
+    let args = text.split(/\s+/);
+
+    // We only support add/remove commands
+    if (!(text.startsWith('add ') || text.startsWith('remove '))) {
+      log(`Unrecognized command`);
+      this.client.say(from, `Command not recognized, boss. ${usage}`);
       return;
     }
 
-    let key = config.channels[channels.indexOf(message.args[1])].key;
+    // Verify command format
+    if (args.length != 4) {
+      log(`Unrecognized command`);
+      this.client.say(from, `Not enough commands. ${usage}`);
```

Comment:
Maybe "incorrect number of arguments"? because it could have more than 4 and still trigger this case.